scripts/zones/Bastok_Markets/npcs/Matthias.lua

```diff
@@ -46,7 +161,64 @@ end;
 function onEventFinish(player,csid,option)
 --printf("CSID: %u",csid);
 --printf("RESULT: %u",option);
-end;
-
-
-
+	
+	local completedPieces = player:getVar("dancerCompletedAF");
+
+	if(csid == 0x01EC) then
+	
+		if(option > 0) then -- option 1 - tiara 2 - bangles 3 - shoes
+			player:setVar("dancerAFChoice", option);
+			player:setVar("dancerTailorCS", 4);
+		else
+			player:setVar("dancerTailorCS", 3); 
+		end;
+		
+	elseif(csid == 0x01ED) then
+	
+		if(option > 0) then -- option 1 - tiara 2 - bangles 3 - shoes
+			
+			local choiceBit = bit.lshift(1, option - 1) --check to see if the player already did this piece already
+
+			if(bit.band(choiceBit, completedPieces) == choiceBit) then
+				player:startEvent(0x01F2); 
+		
+			else
+				player:setVar("dancerAFChoice", option);
+				player:setVar("dancerTailorCS", 4);
+			end;
+		end;
+		
+	elseif(csid == 0x01F1) then -- reward player the appropriate AF
+	
+        local dancerAFID = 1; -- variable used to convert player's choice into an Item ID.
+        local playersAFChoice = player:getVar("dancerAFChoice");
+		
+        if(playersAFChoice == 1) then
+            dancerAFID = dancersTiara;
+        elseif(playersAFChoice == 2) then
+            dancerAFID = dancersBangles;
+        elseif(playersAFChoice == 3) then
+            dancerAFID = dancersToeshoes;
+        end;
+		
+        if(player:getFreeSlotsCount() == 0) then --check to see if the player has enough inventory space before rewarding them.
+            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED, dancerAFID);
```

Comment:
if the inventory is full, the function should immediately return - or else you set the var for having completed the quest, but they won't obtain the item